Please release New Version to solve libiconv build error with GCC 15.2 on Windows

<!--
**Do not report Security Vulnerabilities here**

If you intend to report a security vulnerability, please do so at [HackerOne](https://hackerone.com/nokogiri) following the process detailed in [`SECURITY.md`](https://nokogiri.org/SECURITY.html). Do not report it through GitHub.
-->

**Please describe the bug**

When I update gcc to 15.2 on Windows, libiconv 1.17 causes build error.
This error will be solve with libiconv 1.18.
So, could you please release new nokogiri version ?

### libiconv 1.17 

```
PS C:\> ridk exec pacman -Syuu
PS C:\> ridk exec gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=C:/Ruby33-x64/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../gcc-15.2.0/configure --prefix=/ucrt64 --with-local-prefix=/ucrt64/local --with-native-system-header-dir=/ucrt64/include --libexecdir=/ucrt64/lib --enable-bootstrap --enable-checking=release --with-arch=nocona --with-tune=generic --enable-mingw-wildcard --enable-languages=c,lto,c++,fortran,ada,objc,obj-c++,jit --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-backtrace=yes --enable-libstdcxx-filesystem-ts --enable-libstdcxx-time --disable-libstdcxx-pch --enable-lto --enable-libgomp --disable-libssp --disable-multilib --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/ucrt64 --with-mpfr=/ucrt64 --with-mpc=/ucrt64 --with-isl=/ucrt64 --with-pkgversion='Rev8, Built by MSYS2 project' --with-bugurl=https://github.com/msys2/MINGW-packages/issues --with-gnu-as --with-gnu-ld --with-libstdcxx-zoneinfo=yes --disable-libstdcxx-debug --enable-plugin --with-boot-ldflags=-static-libstdc++ --with-stage1-ldflags=-static-libstdc++
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 15.2.0 (Rev8, Built by MSYS2 project) 
```


```
PS C:\tmp\nokogiri-1.18.9\ext\nokogiri> ridk exec ruby extconf.rb
checking for whether -std=c99 is accepted as CFLAGS... yes
checking for whether -Wno-declaration-after-statement is accepted as CFLAGS... yes
checking for whether -O2 is accepted as CFLAGS... yes
checking for whether -g is accepted as CFLAGS... yes
checking for whether -Winline is accepted as CFLAGS... yes
checking for whether -Wmissing-noreturn is accepted as CFLAGS... yes
checking for whether -Wconversion is accepted as CFLAGS... no
checking for whether  "-Idummypath" is accepted as CPPFLAGS... yes
Building nokogiri using packaged libraries.
Static linking is enabled.
Cross build is disabled.
Using mini_portile version 2.8.9
Building Nokogiri with a packaged version of zlib-1.3.1.
Activating zlib 1.3.1 (from C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/zlib/1.3.1)...
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of libiconv-1.17.
Configuration options: --host\=x86_64-w64-mingw32 --enable-static --disable-shared --libdir\=C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/libiconv/1.17/lib --disable-dependency-tracking --disable-shared --enable-static CPPFLAGS\=-Wall CFLAGS\=-O2\ -g\ -fPIC CXXFLAGS\=-O2\ -g LDFLAGS\=

The Nokogiri maintainers intend to provide timely security updates, but if
this is a concern for you and want to use your OS/distro system library
instead, then abort this installation process and install nokogiri as
instructed at:

  https://nokogiri.org/tutorials/installing_nokogiri.html#installing-using-standard-system-libraries

Extracting libiconv-1.17.tar.gz into tmp/x86_64-w64-mingw32/ports/libiconv/1.17... OK
Running 'compile' for libiconv 1.17... ERROR. Please review logs to see what happened:
----- contents of 'C:/tmp/nokogiri-1.18.9/ext/nokogiri/tmp/x86_64-w64-mingw32/ports/libiconv/1.17/compile.log' -----
cd lib && make all
make[1]: ディレクトリ '/c/tmp/nokogiri-1.18.9/ext/nokogiri/tmp/x86_64-w64-mingw32/ports/libiconv/1.17/libiconv-1.17/lib' に入ります
/bin/sh ../libtool --mode=compile gcc -I. -I. -I../include -I./../include -I.. -I./.. -Wall -O2 -g -fPIC  -DLIBDIR=\"C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/libiconv/1.17/lib\" -DBUILDING_LIBICONV -DBUILDING_LIBCHARSET -DBUILDING_DLL -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/libiconv/1.17/lib\" -DNO_XMALLOC -Dset_relocation_prefix=libiconv_set_relocation_prefix -Drelocate=libiconv_relocate -Drelocate2=libiconv_relocate2 -DHAVE_CONFIG_H -c ./iconv.c
libtool: compile:  gcc -I. -I. -I../include -I./../include -I.. -I./.. -Wall -O2 -g -fPIC -DLIBDIR=\"C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/libiconv/1.17/lib\" -DBUILDING_LIBICONV -DBUILDING_LIBCHARSET -DBUILDING_DLL -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"C:/tmp/nokogiri-1.18.9/ports/x64-mingw-ucrt/libiconv/1.17/lib\" -DNO_XMALLOC -Dset_relocation_prefix=libiconv_set_relocation_prefix -Drelocate=libiconv_relocate -Drelocate2=libiconv_relocate2 -DHAVE_CONFIG_H -c ./iconv.c -o iconv.o
In file included from ./loops.h:23,
                 from ./iconv.c:148:
./loop_wchar.h:39:17: error: conflicting types for 'mbrtowc'; have 'size_t(void)' {aka 'long long unsigned int(void)'}
   39 |   extern size_t mbrtowc ();
      |                 ^~~~~~~
In file included from ../include/iconv.h:118,
                 from ./iconv.c:20:
C:/Ruby33-x64/msys64/ucrt64/include/wchar.h:1212:18: note: previous declaration of 'mbrtowc' with type 'size_t(wchar_t * restrict,  const char * restrict,  size_t,  mbstate_t * restrict)' {aka 'long long unsigned int(short unsigned int * restrict,  const char * restrict,  long long unsigned int,  mbstate_t * restrict)'}
 1212 |   size_t __cdecl mbrtowc(wchar_t * __restrict__ _DstCh,const char * __restrict__ _SrcCh,size_t _SizeInBytes,mbstate_t * __restrict__ _State);
      |                  ^~~~~~~
./loop_wchar.h: In function 'wchar_to_loop_convert':
./loop_wchar.h:356:15: error: too many arguments to function 'mbrtowc'; expected 0, have 4
  356 |         res = mbrtowc(&wc,buf,bufcount,&state);
      |               ^~~~~~~ ~~~
./loop_wchar.h:39:17: note: declared here
   39 |   extern size_t mbrtowc ();
      |                 ^~~~~~~

...(snip)...

```

### libiconv 1.18


```
PS C:\src\nokogiri\ext\nokogiri> ridk exec ruby extconf.rb
checking for whether -std=c99 is accepted as CFLAGS... yes
checking for whether -Wno-declaration-after-statement is accepted as CFLAGS... yes
checking for whether -O2 is accepted as CFLAGS... yes
checking for whether -g is accepted as CFLAGS... yes
checking for whether -Winline is accepted as CFLAGS... yes
checking for whether -Wmissing-noreturn is accepted as CFLAGS... yes
checking for whether -Wconversion is accepted as CFLAGS... no
checking for whether  "-Idummypath" is accepted as CPPFLAGS... yes
Building nokogiri using packaged libraries.
Static linking is enabled.
Cross build is disabled.
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of zlib-1.3.1.
Configuration options: --host\=x86_64-w64-mingw32 --enable-static --disable-shared --libdir\=C:/src/nokogiri/ports/x64-mingw-ucrt/zlib/1.3.1/lib --disable-shared --enable-static CFLAGS\=-fPIC

The Nokogiri maintainers intend to provide timely security updates, but if
this is a concern for you and want to use your OS/distro system library
instead, then abort this installation process and install nokogiri as
instructed at:

  https://nokogiri.org/tutorials/installing_nokogiri.html#installing-using-standard-system-libraries

Downloading zlib-1.3.1.tar.gz (100%) 
Extracting zlib-1.3.1.tar.gz into tmp/x86_64-w64-mingw32/ports/zlib/1.3.1... OK
Running 'compile' for zlib 1.3.1... OK
Running 'install' for zlib 1.3.1... OK
Activating zlib 1.3.1 (from C:/src/nokogiri/ports/x64-mingw-ucrt/zlib/1.3.1)...
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of libiconv-1.18.
Configuration options: --host\=x86_64-w64-mingw32 --enable-static --disable-shared --libdir\=C:/src/nokogiri/ports/x64-mingw-ucrt/libiconv/1.18/lib --disable-dependency-tracking --disable-shared --enable-static CPPFLAGS\=-Wall CFLAGS\=-O2\ -g\ -fPIC CXXFLAGS\=-O2\ -g LDFLAGS\=

The Nokogiri maintainers intend to provide timely security updates, but if
this is a concern for you and want to use your OS/distro system library
instead, then abort this installation process and install nokogiri as
instructed at:

  https://nokogiri.org/tutorials/installing_nokogiri.html#installing-using-standard-system-libraries

2 retrie(s) left for libiconv-1.18.tar.gz (Net::OpenTimeout)
Downloading libiconv-1.18.tar.gz (100%) 
Extracting libiconv-1.18.tar.gz into tmp/x86_64-w64-mingw32/ports/libiconv/1.18... OK
Running 'configure' for libiconv 1.18... C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:603: warning: Exception in finalizer #<Tempfile::Remover:0x00000216181d75d8 @pid=19728, @path="C:/Users/watso/AppData/Local/Temp/download-libiconv-1.18.tar.gz20250823-19728-nld0m0">
C:/Ruby33-x64/lib/ruby/3.3.0/tempfile.rb:319:in `unlink': Permission denied @ apply2files - C:/Users/watso/AppData/Local/Temp/download-libiconv-1.18.tar.gz20250823-19728-nld0m0 (Errno::EACCES)
	from C:/Ruby33-x64/lib/ruby/3.3.0/tempfile.rb:319:in `call'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:603:in `each'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:603:in `spawn'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:603:in `block in execute'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:594:in `chdir'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:594:in `execute'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:191:in `configure'
	from C:/Ruby33-x64/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.9/lib/mini_portile2/mini_portile.rb:237:in `cook'
	from extconf.rb:550:in `block (2 levels) in process_recipe'
	from extconf.rb:329:in `chdir'
	from extconf.rb:329:in `chdir_for_build'
	from extconf.rb:550:in `block in process_recipe'
	from <internal:kernel>:90:in `tap'
	from extconf.rb:448:in `process_recipe'
	from extconf.rb:824:in `<main>'
OK
Running 'compile' for libiconv 1.18... OK
Running 'install' for libiconv 1.18... OK
Activating libiconv 1.18 (from C:/src/nokogiri/ports/x64-mingw-ucrt/libiconv/1.18)...
checking for whether -IC:/src/nokogiri/ports/x64-mingw-ucrt/zlib/1.3.1/include is accepted as CPPFLAGS... yes
checking for gzdopen() in -lz... yes
checking for whether -IC:/src/nokogiri/ports/x64-mingw-ucrt/libiconv/1.18/include is accepted as CPPFLAGS... yes
checking for iconv_open() in -liconv... yes
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of libxml2-2.14.5.
Configuration options: --host\=x86_64-w64-mingw32 --enable-static --disable-shared --libdir\=C:/src/nokogiri/ports/x64-mingw-ucrt/libxml2/2.14.5/lib --with-legacy --with-zlib\=C:/src/nokogiri/ports/x64-mingw-ucrt/zlib/1.3.1 --with-iconv\=C:/src/nokogiri/ports/x64-mingw-ucrt/libiconv/1.18 --disable-dependency-tracking --without-python --without-readline --with-c14n --with-debug --with-threads --disable-shared --enable-static CPPFLAGS\= CFLAGS\=-O2\ -g\ -ULIBXML_STATIC\ -DIN_LIBXML\ -fPIC
The following patches are being applied:
  - 0001-Remove-script-macro-support.patch
  - 0002-Update-entities-to-remove-handling-of-ssi.patch
  - 0009-allow-wildcard-namespaces.patch
  - 0010-update-config.guess-and-config.sub-for-libxml2.patch

The Nokogiri maintainers intend to provide timely security updates, but if
this is a concern for you and want to use your OS/distro system library
instead, then abort this installation process and install nokogiri as
instructed at:

  https://nokogiri.org/tutorials/installing_nokogiri.html#installing-using-standard-system-libraries

Note, however, that nokogiri cannot guarantee compatibility with every
version of libxml2 that may be provided by OS/package vendors.

Downloading libxml2-2.14.5.tar.xz (100%) 
Extracting libxml2-2.14.5.tar.xz into tmp/x86_64-w64-mingw32/ports/libxml2/2.14.5... OK
Running patch with C:/src/nokogiri/patches/libxml2/0001-Remove-script-macro-support.patch... OK
Running patch with C:/src/nokogiri/patches/libxml2/0002-Update-entities-to-remove-handling-of-ssi.patch... OK
Running patch with C:/src/nokogiri/patches/libxml2/0009-allow-wildcard-namespaces.patch... OK
Running patch with C:/src/nokogiri/patches/libxml2/0010-update-config.guess-and-config.sub-for-libxml2.patch... OK
Running 'configure' for libxml2 2.14.5... OK
Running 'compile' for libxml2 2.14.5... OK
Running 'install' for libxml2 2.14.5... OK
Activating libxml2 2.14.5 (from C:/src/nokogiri/ports/x64-mingw-ucrt/libxml2/2.14.5)...
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of libxslt-1.1.43.
Configuration options: --host\=x86_64-w64-mingw32 --enable-static --disable-shared --libdir\=C:/src/nokogiri/ports/x64-mingw-ucrt/libxslt/1.1.43/lib --disable-dependency-tracking --without-python --without-crypto --with-debug --with-libxml-prefix\=/C/src/nokogiri/ports/x64-mingw-ucrt/libxml2/2.14.5 --disable-shared --enable-static CFLAGS\=-O2\ -g\ -ULIBXSLT_STATIC\ -DIN_LIBXSLT\ -ULIBEXSLT_STATIC\ -DIN_LIBEXSLT\ -fPIC
The following patches are being applied:
  - 0001-update-config.guess-and-config.sub-for-libxslt.patch

The Nokogiri maintainers intend to provide timely security updates, but if
this is a concern for you and want to use your OS/distro system library
instead, then abort this installation process and install nokogiri as
instructed at:

  https://nokogiri.org/tutorials/installing_nokogiri.html#installing-using-standard-system-libraries

Downloading libxslt-1.1.43.tar.xz (100%) 
Extracting libxslt-1.1.43.tar.xz into tmp/x86_64-w64-mingw32/ports/libxslt/1.1.43... OK
Running patch with C:/src/nokogiri/patches/libxslt/0001-update-config.guess-and-config.sub-for-libxslt.patch... OK
Running 'configure' for libxslt 1.1.43... OK
Running 'compile' for libxslt 1.1.43... OK
Running 'install' for libxslt 1.1.43... OK
Activating libxslt 1.1.43 (from C:/src/nokogiri/ports/x64-mingw-ucrt/libxslt/1.1.43)...
checking for whether -DNOKOGIRI_PACKAGED_LIBRARIES is accepted as CPPFLAGS... yes
xml2-config cflags: -IC:/src/nokogiri/ports/x64-mingw-ucrt/libxml2/2.14.5/include/libxml2 -DLIBXML_STATIC
checking for whether -DNOKOGIRI_LIBXML2_PATCHES="\"0001-Remove-script-macro-support.patch 0002-Update-entities-to-remove-handling-of-ssi.patch 0009-allow-wildcard-namespaces.patch 0010-update-config.guess-and-config.sub-for-libxml2.patch\"" is accepted as CPPFLAGS... yes
checking for -llzma... no
checking for -lliblzma... no
xslt-config cflags: -IC:/src/nokogiri/ports/x64-mingw-ucrt/libxml2/2.14.5/include/libxml2 -DLIBXML_STATIC -DLIBXSLT_STATIC -DLIBEXSLT_STATIC -IC:/src/nokogiri/ports/x64-mingw-ucrt/libxslt/1.1.43/include
checking for whether -DNOKOGIRI_LIBXSLT_PATCHES="\"0001-update-config.guess-and-config.sub-for-libxslt.patch\"" is accepted as CPPFLAGS... yes
checking for xmlParseDoc() in libxml/parser.h... yes
checking for xsltParseStylesheetDoc() in libxslt/xslt.h... yes
checking for exsltFuncRegister() in libexslt/exslt.h... yes
Using mini_portile version 2.8.9
---------- IMPORTANT NOTICE ----------
Building Nokogiri with a packaged version of libgumbo-1.0.0-nokogiri.
Configuration options: --disable-shared --enable-static CFLAGS\=-fPIC
Copying gumbo-parser files into tmp/x86_64-w64-mingw32/ports/libgumbo/1.0.0-nokogiri/gumbo-parser...
Running 'compile' for libgumbo 1.0.0-nokogiri... OK
Activating libgumbo 1.0.0-nokogiri (from ports/x64-mingw-ucrt/libgumbo/1.0.0-nokogiri)...
checking for whether -IC:/src/nokogiri/ext/nokogiri/ports/x64-mingw-ucrt/libgumbo/1.0.0-nokogiri/include is accepted as CPPFLAGS... yes
checking for gumbo_parse_with_options() in nokogiri_gumbo.h... yes
checking for xmlCtxtSetOptions()... yes
checking for xmlCtxtGetOptions()... yes
checking for xmlSwitchEncodingName()... yes
checking for xmlAddIDSafe()... yes
checking for rb_category_warning()... yes
checking for whether -DNOKOGIRI_OTHER_LIBRARY_VERSIONS="\"zlib:1.3.1,libiconv:1.18,libgumbo:1.0.0-nokogiri\"" is accepted as CPPFLAGS... yes
creating Makefile
```


**Help us reproduce what you're seeing**

<!--
If possible, please include a complete, self-contained script that reproduces the behavior you're seeing. Please try to remove external dependencies like Rails or other libraries that may be wrapping Nokogiri.

Here's an example of how you might structure such a script:

```ruby
#! /usr/bin/env ruby

require 'nokogiri'
require 'minitest/autorun'

class Test < Minitest::Spec
  describe "Node#css" do
    it "should find a div using chained classes" do
      html = <<~HEREDOC
        <html>
          <body>
            <div class="foo"> one</div> 
            <div class="bar">two</div> 
            <div class="foo bar">three</div> 
      HEREDOC
      
      doc = Nokogiri::HTML::Document.parse(html)
      
      assert_equal 1, doc.css("div.foo.bar").length
      assert_equal "three", doc.at_css("div.foo.bar").text
    end
  end
end
```

If you haven't included a test, please provide whatever code you can, the inputs (if any), along with the output that you're seeing. We need to reproduce what you're seeing to be able to help!
-->


**Expected behavior**

<!--
If you haven't included a test above, please tell us precisely what you expected to happen.
-->


**Environment**

<!--
```
Please paste the output from `nokogiri -v` here, escaped by triple-backtick.
```

This output will tell us what version of Ruby you're using, how you installed nokogiri, what versions of the underlying libraries you're using, and what operating system you're using.
-->

**Additional context**

<!--
Add any other context about the problem here.
-->

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -271,7 +271,9 @@ jobs:
       matrix:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby) }}
-    runs-on: macos-latest
+        include:
+          - { ruby: "3.1", os: macos-14 }
+    runs-on: ${{ matrix.os || 'macos-latest' }}
     steps:
       - uses: actions/checkout@v4
         with:
@@ -526,7 +528,9 @@ jobs:
       matrix:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby) }}
-    runs-on: macos-latest
+        include:
+          - { ruby: "3.1", os: macos-14 }
+    runs-on: ${{ matrix.os || 'macos-latest' }}
     steps:
       - uses: actions/checkout@v4
         with:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -328,6 +328,9 @@ jobs:
       matrix:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby_win) }}
+        exclude:
+          - sys: "enable"
+            ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
     runs-on: windows-2022
     steps:
       - name: configure git crlf
@@ -340,7 +343,7 @@ jobs:
       - uses: ruby/setup-ruby-pkgs@v1
         with:
           ruby-version: "${{matrix.ruby}}"
-          mingw: "libxml2 libxslt"
+          mingw: ${{ matrix.ruby == '3.1' && ' ' || 'libxml2 libxslt' }} # '' is falsey, ' ' is truthy
           bundler-cache: true
           bundler: latest
       - uses: actions/cache@v4
@@ -551,6 +554,9 @@ jobs:
       matrix:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby_win) }}
+        exclude:
+          - sys: "enable"
+            ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
     runs-on: windows-2022
     steps:
       - uses: actions/checkout@v4
@@ -559,7 +565,7 @@ jobs:
       - uses: ruby/setup-ruby-pkgs@v1
         with:
           ruby-version: "${{matrix.ruby}}"
-          mingw: "libxml2 libxslt"
+          mingw: ${{ matrix.ruby == '3.1' && ' ' || 'libxml2 libxslt' }} # '' is falsey, ' ' is truthy
       - uses: actions/download-artifact@v4
         with:
           name: generic-gem

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -329,8 +329,7 @@ jobs:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby_win) }}
         exclude:
-          - sys: "enable"
-            ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
+          - ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
     runs-on: windows-2022
     steps:
       - name: configure git crlf
@@ -343,7 +342,7 @@ jobs:
       - uses: ruby/setup-ruby-pkgs@v1
         with:
           ruby-version: "${{matrix.ruby}}"
-          mingw: ${{ matrix.ruby == '3.1' && ' ' || 'libxml2 libxslt' }} # '' is falsey, ' ' is truthy
+          mingw: "libxml2 libxslt"
           bundler-cache: true
           bundler: latest
       - uses: actions/cache@v4
@@ -555,8 +554,7 @@ jobs:
         sys: ["enable", "disable"]
         ruby: ${{ fromJSON(needs.ruby_versions.outputs.setup_ruby_win) }}
         exclude:
-          - sys: "enable"
-            ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
+          - ruby: "3.1" # because Ruby 3.1 devkit is built with a version of GCC too old for modern msys2 2025-08
     runs-on: windows-2022
     steps:
       - uses: actions/checkout@v4
@@ -565,7 +563,7 @@ jobs:
       - uses: ruby/setup-ruby-pkgs@v1
         with:
           ruby-version: "${{matrix.ruby}}"
-          mingw: ${{ matrix.ruby == '3.1' && ' ' || 'libxml2 libxslt' }} # '' is falsey, ' ' is truthy
+          mingw: "libxml2 libxslt"
       - uses: actions/download-artifact@v4
         with:
           name: generic-gem

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -662,7 +662,7 @@ jobs:
           name: cruby-${{ matrix.platform }}-gem
           path: gems
       - run: |
-          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+          docker run --rm --privileged alekitto/qemu-user-static --reset -p yes
           docker run --rm -v $PWD:/nokogiri -w /nokogiri \
             ${{ matrix.docker_platform }} ruby:${{ matrix.ruby }}${{ matrix.docker_tag }} \
             sh -c "

--- a/dependencies.yml
+++ b/dependencies.yml
@@ -15,28 +15,17 @@ zlib:
   # SHA-256 hash provided on http://zlib.net/
 
 libiconv:
-  version: "1.17"
-  sha256: "8f74213b56238c85a50a5329f77e06198771e70dd9a739779f4c02f65d971313"
-  # signature verified by following this path:
-  # - release announced at https://savannah.gnu.org/forum/forum.php?forum_id=10175
-  # - which links to https://savannah.gnu.org/users/haible as the releaser
-  # - which links to https://savannah.gnu.org/people/viewgpg.php?user_id=1871 as the gpg key
-  #
-  # So:
-  # - wget -q -O - https://savannah.gnu.org/people/viewgpg.php?user_id=1871 | gpg --import
-  #     gpg: key F5BE8B267C6A406D: 1 signature not checked due to a missing key
+  # $ gpg --keyserver keyserver.ubuntu.com --recv 9001B85AF9E1B83DF1BDA942F5BE8B267C6A406D
   #     gpg: key F5BE8B267C6A406D: public key "Bruno Haible (Open Source Development) <bruno@clisp.org>" imported
   #     gpg: Total number processed: 1
   #     gpg:               imported: 1
-  #     gpg: marginals needed: 3  completes needed: 1  trust model: pgp
-  #     gpg: depth: 0  valid:   4  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 4u
-  #     gpg: next trustdb check due at 2024-05-09
-  # - gpg --verify libiconv-1.17.tar.gz.sig ports/archives/libiconv-1.17.tar.gz
-  #     gpg: Signature made Sun 15 May 2022 11:26:42 AM EDT
+  # $ gpg --verify libiconv-1.18.tar.gz.sig ports/archives/libiconv-1.18.tar.gz
+  #     gpg: Signature made Sun 15 Dec 2024 07:26:18 AM EST
   #     gpg:                using RSA key 9001B85AF9E1B83DF1BDA942F5BE8B267C6A406D
-  #     gpg: Good signature from "Bruno Haible (Open Source Development) <bruno@clisp.org>" [unknown]
-  #     gpg: WARNING: This key is not certified with a trusted signature!
-  #     gpg:          There is no indication that the signature belongs to the owner.
+  #     gpg: Good signature from "Bruno Haible (Open Source Development) <bruno@clisp.org>" [expired]
+  #     gpg: Note: This key has expired!
   #     Primary key fingerprint: 9001 B85A F9E1 B83D F1BD  A942 F5BE 8B26 7C6A 406D
-  #
-  # And this sha256sum is calculated from that verified tarball.
+  # $ sha256sum ports/archives/libiconv-1.18.tar.gz
+  #     3b08f5f4f9b4eb82f151a7040bfd6fe6c6fb922efe4b1659c66ea933276965e8  ports/archives/libiconv-1.18.tar.gz
+  version: "1.18"
+  sha256: "3b08f5f4f9b4eb82f151a7040bfd6fe6c6fb922efe4b1659c66ea933276965e8"

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -832,7 +832,7 @@ else
         cross_build_p,
       ) do |recipe|
         recipe.files = [{
-          url: "https://ftp.gnu.org/pub/gnu/libiconv/#{recipe.name}-#{recipe.version}.tar.gz",
+          url: "https://ftpmirror.gnu.org/gnu/libiconv/#{recipe.name}-#{recipe.version}.tar.gz",
           sha256: dependencies["libiconv"]["sha256"],
         }]
 

